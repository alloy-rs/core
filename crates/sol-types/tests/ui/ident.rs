use alloy_sol_types::sol;

sol! {
    struct Reserved {
        bool uint256;
    }
}

sol! {
    struct NotAllowed {
        bool è;
    }
}

sol! {
    struct NotAllowedRaw {
        bool r#è;
    }
}

sol! {
    struct Allowed {
        bool r#uint256;
        // bool r#$dollars$;
        bool _underscores_;
        // bool $_$_$_;
    }
}

fn main() {}
