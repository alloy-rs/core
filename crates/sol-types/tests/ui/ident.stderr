error: expected identifier, found reserved keyword
 --> tests/ui/ident.rs:5:14
  |
5 |         bool uint256;
  |              ^^^^^^^

error: invalid identifier
  --> tests/ui/ident.rs:11:14
   |
11 |         bool Ã¨;
   |              ^

error: invalid identifier
  --> tests/ui/ident.rs:17:14
   |
17 |         bool r#Ã¨;
   |              ^^^
